File: Shared/Blog.Core/Infrastructure/Middlewares/ApplicationBuilderExtensions.cs
// <copyright file="ApplicationBuilderExtensions.cs" company="PlaceholderCompany">
// Copyright (c) PlaceholderCompany. All rights reserved.
// </copyright>

namespace Blog.Core.Infrastructure.Middlewares;

using Microsoft.AspNetCore.Builder;

/// <summary>
/// ETagger extension.
/// </summary>
public static class ApplicationBuilderExtensions
{
    /// <summary>
    /// Uses the e tagger.
    /// </summary>
    /// <param name="app">The application.</param>
    public static void UseETagger(this IApplicationBuilder app)
    {
        app.UseMiddleware<ETagMiddleware>();
    }
}

File: Shared/Blog.FSharp.Core/Infrastructure/Middlewares/ApplicationBuilderExtensions.fs
namespace Blog.FSharp.Core.Infrastructure.Middlewares

open Microsoft.AspNetCore.Builder

[<AutoOpen>]
module ApplicationBuilderExtensions =

    type IApplicationBuilder with
        /// <summary>
        /// Uses the e tagger.
        /// </summary>
        member this.UseETagger() =
            this.UseMiddleware<ETagMiddleware>() |> ignore